--- dosbox-0.72.orig/src/gui/sdlmain.cpp
+++ dosbox-0.72/src/gui/sdlmain.cpp
@@ -1061,6 +1061,10 @@
 	if (!sdl.mouse.autoenable) SDL_ShowCursor(SDL_DISABLE);
 	sdl.mouse.autolock=false;
 	sdl.mouse.sensitivity=section->Get_int("sensitivity");
+
+	/* Setup Mouse correctly if fullscreen */
+	if(sdl.desktop.fullscreen) GFX_CaptureMouse();
+
 	const char * output=section->Get_string("output");
 	if (!strcasecmp(output,"surface")) {
 		sdl.desktop.want_type=SCREEN_SURFACE;